modules/apigee_edge_teams/src/Entity/TeamAppRouteProvider.php

```diff
@@ -118,6 +138,87 @@ protected function getCollectionRouteByTeam(EntityTypeInterface $entity_type) {
     }
   }
 
+  /**
+   * Gets the generate-credential-form route for a team app.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type.
+   *
+   * @return \Symfony\Component\Routing\Route|null
+   *   The generated route, if available.
+   */
+  protected function getGenerateCredentialRoute(EntityTypeInterface $entity_type) {
+    if ($entity_type->hasLinkTemplate('generate-credential-form')) {
+      $route = new Route($entity_type->getLinkTemplate('generate-credential-form'));
+      $route->setDefault('_form', TeamAppCredentialGenerateForm::class);
```

Comment:
I think new permissions should be introduced for these new operations, like not every developer or team member should be able to destroy an existing/request new a API key.

Reply:
This is already being discussed. Thanks @mxr576